clients/src/test/java/org/apache/kafka/clients/consumer/ConsumerRecordsTest.java

```diff
@@ -35,28 +38,109 @@ public class ConsumerRecordsTest {
 
     @Test
     public void iterator() throws Exception {
+        String topic = "topic";
+        int recordSize = 10;
+        int partitionSize = 15;
+        int emptyPartitionInterval = 3;
```

Comment:
Could we define specific partition to be empty? that will get simplified I think